File: pe/visualstudio/PE.FSharp.Tests/Problem0009.Tests.fs
namespace PE.Problem9.Tests

open PE.FSharp.Problem9.ProductOfPythagoreanTriplet
open Microsoft.VisualStudio.TestTools.UnitTesting

module Problem0009Tests = 
  
    [<TestClass>]
    type Problem9() = 

        [<TestMethod>]
        member x.Problem0009_ProductOfPythagoreanTriplet_sumOfTriplet_'1_2_3'_Should_Equal_6() =
            Assert.AreEqual(6L, sumOfTriplet (1L,2L,3L))

        [<TestMethod>]
        member x.Problem0009_ProductOfPythagoreanTriplet_sumOfTriplet_'100_2000_30000'_Should_Equal_32100() =
            Assert.AreEqual(32100L, sumOfTriplet (100L, 2000L, 30000L))

        [<TestMethod>]
        member x.Problem0009_ProductOfPythagoreanTriplet_findFirstTripletThatSumsTo_12_Should_Equal_'3_4_5'() =
            Assert.AreEqual((3L,4L,5L), findFirstTripletThatSumsTo 12L)

        [<TestMethod>]
        member x.Problem0009_ProductOfPythagoreanTriplet_productOfTripletThatSumsTo_12_Should_Equal_60() =
            Assert.AreEqual(60L, productOfTripletThatSumsTo 12L)

        [<TestMethod>]
        member x.Problem0009_ProductOfPythagoreanTriplet_answer() =
            Assert.AreEqual(31875000L, answer)

File: pe/visualstudio/PE.CSharp.Tests/Problem0009.Tests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace PE.CSharp.Tests
{
    [TestClass]
    public class Problem0009Tests
    {
        private static Problem0009 _problem0009;
        
        [ClassInitialize]
        public static void InitializeClass(TestContext ctx)
        {
            _problem0009 = new Problem0009();
        }

        [TestMethod]
        public void Problem0009_FindFirstTripleThatSumsTo_12_Should_Return_3_4_5()
        {
            var expected = new Tuple<long, long, long>(3L, 4L, 5L);

            FindFirstAssert(expected, 12);
        }

        [TestMethod]
        public void Problem0009_FindFirstTripleThatSumsTo_330_Should_Return_105_88_137()
        {
            var expected = new Tuple<long, long, long>(105, 88, 137);

            FindFirstAssert(expected, 330);
        }

        private void FindFirstAssert(Tuple<long,long,long> expected, int input)
        {
            var result = _problem0009.FindFirstTripleThatSumsTo(input);

            Assert.AreEqual(expected, result);
        }

        [TestMethod]
        public void Problem0009_ProductOfTripleThatSumsTo_330_Should_Return_1265880()
        {
            ProductOfTripleAssert(1265880, 330);
        }

        [TestMethod]
        public void Problem0009_ProductOfTripleThatSumsTo_12_Should_Return_60()
        {
            ProductOfTripleAssert(60, 12);
        }

        [TestMethod]
        public void Problem0009_Answer()
        {
            ProductOfTripleAssert(31875000, 1000);
        }

        private void ProductOfTripleAssert(int expected, int input)
        {
            Assert.AreEqual(_problem0009.ProductOfTripleThatSumsTo(input), expected);
        }
    }
}
